File: Algorithm/Python/121_Best Time to Buy and Sell Stock/121_Best Time to Buy and Sell Stock.py
prices = [7, 1, 5, 3, 6, 4]
max_profit = 0
buy = prices[0]

for i in range(1, len(prices)):
    if(max_profit < prices[i] - buy):
        max_profit = prices[i] - buy
    if(prices[i] < buy):
        buy = prices[i]
print(max_profit)

File: Algorithm/C++/121 Best Time to Buy and Sell Stock/121_BestTimetoBuyandSellStock.cpp
class Solution
{
public:
	int maxProfit(vector<int> &prices)
	{
		int size = prices.size();
		int max_profit = 0;
		int buy = prices[0];
		for (int i = 1; i < size; i++)
		{
			if (buy > prices[i])
				buy = prices[i];
			else if (max_profit < prices[i] - buy)
				max_profit = prices[i] - buy;
		}
		return max_profit;
	}
};